src/memory_dataset.jl

```diff
@@ -70,10 +70,24 @@ end
 
 Base.parent(v::MemoryVariable) = v.data
 Base.size(v::MemoryVariable) = size(parent(v))
-Base.getindex(v::MemoryVariable,ij::TIndices...) = parent(v)[ij...]
-function Base.setindex!(v::MemoryVariable,data,ij...)
+
+function DiskArrays.readblock!(v::MemoryVariable{T, N},
+    aout,
+    indexes::Vararg{OrdinalRange, N}) where {T, N}
+
+    aout .= parent(v)[indexes...]
+end
+
+
+function DiskArrays.writeblock!(v::MemoryVariable{T, 0}, data, indexes::Vararg{OrdinalRange, 0}) where {T}
+    parent(v)[] = data[]
```

Comment:
I had to add this to assign scalar values. I wonder if all the packages implementing `CommonDataModel` will have to do the same.

Reply:
Isn't this part redundant? `, indexes::Vararg{OrdinalRange, 0})` I think you can just delete it?

But I'm not sure why its needed, whats the error without it?

Reply:
See comment it latest code where I tagged you.

Reply:
Ahh, did you mean the input argument `indexes::Vararg{OrdinalRange, 0}`. Yes I don't think that is needed. I still need to handle the special case of N==0 explicitly in one way or another.